optica.rb

```diff
@@ -14,7 +14,7 @@ class Optica < Sinatra::Base
   # endpoint for fab usage
   get '/roles' do
     # keys must be strings
-    return get_nodes(request, ['role', 'id', 'hostname'])
+    return get_nodes(request, ['role', 'id', 'hostname', 'roles', 'last_reported', 'environment'])
```

Comment:
we can. Optica treats any query param as a filter argument so I'll need to whitelist it to not do that.